### ISSUE TITLE
Texture coordinate precision

### ISSUE THREAD

**Issue by kvark** [servo/webrender#1398]
We've got issues with interpolated texture coordinates on [images](https://bugzilla.mozilla.org/show_bug.cgi?id=1370908) and gradients (#966). Need a solid way to avoid precision errors there.

Proposed ~~solutions~~ tricks to try out:
  1. interpolate absolute coordinates, divide in the PS
  2. force `highp` on the values involved
  3. derive the texture coordinate in the PS from the fragment coordinate

**Comment by staktrace** [comment]
Ethan [said](https://bugzilla.mozilla.org/show_bug.cgi?id=1389000#c33) that this issue might be the cause of the high number of intermittent failures we get when we try and enable layers-free by default in gecko. Can we bump this up in priority to see if fixing it helps?

**Comment by glennw** [comment]
Is it possible to reproduce these issues in a wrench / yaml test case?

**Comment by kvark** [comment]
@glennw I think we can make one, I'll look into it.

**Comment by kvark** [comment]
Here is the product of `./headless.py png benchmarks/unaligned_gradient.yaml`:
![unaligned-gradient](https://user-images.githubusercontent.com/107301/31420641-c0a41d76-ae10-11e7-9de0-f461733d1974.png)

If you click on it and zoom in, there are horizontal lines visible, which manifest the issue.
When I looked into it last time, the only solution that worked for me was deriving the texture coordinates from `gl_FragCoord`. Hence, I suspected HW interpolation (of user texture coordinates) to be the problem.
